### PR TITLE
Read me update

### PR DESCRIPTION
**Pull request by jakobuid**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,7 +75,7 @@ docs: https://docs.walletconnect.com/2.0/protocol/glossary#controller
 ```kotlin
 val walletDelegate = object : WalletConnectClient.WalletDelegate {
     override fun onSessionProposal(sessionProposal: WalletConnect.Model.SessionProposal) {
-        // Session Proposal object sent by Dapp after pairing was successful
+        // Session Proposal object sent by Dapp
     }
 
     override fun onSessionRequest(sessionRequest: WalletConnect.Model.SessionRequest) {
```
Comment (Elyniss) [inline README.md:81]: I would change comment in this method to
`    // Triggered when a Dapp sends SessionRequest to sign a transaction or a message`

File: README.md [hunk 2]
```diff
@@ -107,179 +117,145 @@ the Dapp.
...
+val disconnectionReason: String = /*The reason for disconnecting the Session*/
+val disconnectionCode: String = /*The code for for disconnecting the Session*/
+val sessionTopic: String = /*Topic from the Session*/
+val disconnectParams = WalletConnect.Params.Disconnect(sessionTopic, disconnectionReason, disconnectionCode)
 
-WalletConnectClient.disconnect(disconnectParams, listener)
+WalletConnectClient.disconnect(disconnectParams)
 ```
-To disconnect from a settle session, pass a disconnection reason and the Settled Session topic to the `WalletConnectClient.disconnect` function. Listener will asynchronously expose a DeleteSession object that will mirror the data sent for rejection.
+To disconnect from a settle session, pass a disconnection reason with code and the Session topic to the `WalletConnectClient.disconnect` function.
```
Comment (Elyniss) [inline README.md:167]:
```suggestion
To disconnect from a settled session, pass a disconnection reason with code and the Session topic to the `WalletConnectClient.disconnect` function.
```

File: README.md [hunk 3]
```diff
@@ -107,179 +117,145 @@ the Dapp.
...
-   override fun onSuccess(deletedSession: WalletConnect.Model.DeletedSession) {
-      // DeleteSession object with topic and reason
-   }
-
-   override fun onError(error: Throwable) {
-      // Session disconnect error
-   }
-}
+val disconnectionReason: String = /*The reason for disconnecting the Session*/
+val disconnectionCode: String = /*The code for for disconnecting the Session*/
```
Comment (Elyniss) [inline README.md:161]:
```suggestion
val disconnectionCode: String = /*The code for for disconnecting the Session. Follow [the link](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-25.md) for standard codes*/
```

File: README.md [hunk 4]
```diff
@@ -107,179 +117,145 @@ the Dapp.
...
-val listener = object : WalletConnect.Listeners.SessionDelete {
-   override fun onSuccess(deletedSession: WalletConnect.Model.DeletedSession) {
-      // DeleteSession object with topic and reason
-   }
-
-   override fun onError(error: Throwable) {
-      // Session disconnect error
-   }
-}
+val disconnectionReason: String = /*The reason for disconnecting the Session*/
```
Comment (Elyniss) [inline README.md:160]:
```suggestion
val disconnectionReason: String = /*The reason for disconnecting the Session. Follow [the link](Follow [the link](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-25.md) for standard codes) for standard reasons*/
```

File: README.md [hunk 5]
```diff
@@ -55,23 +54,18 @@ val init = WalletConnect.Params.Init(
...
-   isController = /*true or false*/, 
-   projectId = "project id", 
    appMetaData = appMetaData
 )
 
 WalletConnectClient.initalize(init)
 ```
 
-The controller client will always be the wallet which is exposing blockchain accounts to a Dapp and therefore is also in charge of signing. To
+The wallet client will always be responsible of exposing accounts(CAPI10) to a Dapp and therefore is also in charge of signing. To
```
Comment (TalhaAli00) [inline README.md:63]: `The wallet client will always be responsible for exposing`

File: README.md [hunk 6]
```diff
@@ -107,179 +117,146 @@ the Dapp.
...
 
-   override fun onError(error: Throwable) {
-      // Error
-   }
-}
+WalletConnectClient.updateAccounts(updateParams) { error -> /*optional callback for error while sending update accounts request*/ }
+```
+To update a session with accounts, create a `WalletConnect.Params.UpdateAccounts` object with the session's topic and accounts to update session with to `WalletConnectClient.updateAccounts`.
+
+&nbsp;
```
Comment (TalhaAli00) [inline README.md:204]: remove

**Review by TalhaAli00 — APPROVED** [review]

**Pull request merged**